custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -303,7 +303,9 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
 
             // popup support on rankings pages
             vm.displayMobilePopup = function (location) {
-                var locationData = vm.data.mapData.data[location.loc_name];
+                // data is stored in .data for the first three location levels, but then moves to tooltips_data
```

Comment:
out of curiosity, why does it move like this?

Reply:
I'd guess that the answer is because we don't have maps below that level, so putting something in `mapData` felt weird, and putting something in `tooltips_data` more accurately reflects how it's used on the chart page. But basically all the reports [do something like this](https://github.com/dimagi/commcare-hq/blob/66b321e663a586a13f51ac641031bd3d996b5379/custom/icds_reports/views.py#L523-L526)

To me it feels weird that python is handling the formatting of this data for how it ends being used on the page instead of e.g. serving up the raw data and letting JS handle the transformation. But that's a gripe for another time.